mobile/library/cc/engine_builder.cc

```diff
@@ -303,6 +314,16 @@ std::string EngineBuilder::generateConfigStr() const {
     insertCustomFilter(filter_config, config_template);
   }
 
+  if (!custom_layers_.empty()) {
+    absl::StrReplaceAll({{"#{custom_layers}", absl::StrCat("#{custom_layers}\n", custom_layers_)}},
+                        &config_template);
+  }
+
+  if (!custom_ads_.empty()) {
+    absl::StrReplaceAll({{"#{custom_ads}", absl::StrCat("#{custom_ads}\n", custom_ads_)}},
```

Comment:
before setting custom_layers_, we should set custom_ads_, and then before setting custom_layers_, check that custom_ads_ is set, and if not, throw a std::runtime_error